code/modules/admin/bccm/bccm.dm

```diff
@@ -0,0 +1,387 @@
+// BCCM (Ban Counter Counter Measures system) ((Name subject to change)), originally inspired by EAMS (Epic Anti-Multiaccount System), by Epicus
+//version 1.0.2
+
+/datum/bccm_info
+	var/is_loaded = FALSE
+	var/is_whitelisted = FALSE
+
+	var/ip
+	var/ip_as
+	var/ip_mobile
+	var/ip_proxy
+	var/ip_hosting
+
+/client
+	var/datum/bccm_info/bccm_info = new
+
+SUBSYSTEM_DEF(bccm)
+	name = "BCCM"
+	init_order = SS_INIT_BCCM
+	flags = SS_NO_FIRE
+
+	var/max_error_count = 4
+
+	var/is_active = FALSE
+	var/error_counter = 0
+
+	var/list/tgui_panel_asn_data = list()
+	var/list/tgui_panel_wl_data = list()
+
+	var/list/client/postponed_client_queue = new
+
+/datum/controller/subsystem/bccm/Initialize(timeofday)
+	if(!config.bccm)
+		return ..()
+
+	if(!sqlenabled)
+		log_debug("BCCM could not be loaded without SQL enabled")
+		return ..()
+
+	Toggle()
+	return ..()
+
+/datum/controller/subsystem/bccm/stat_entry(msg)
+	if(is_active)
+		msg = "ACTIVE"
+	else
+		msg = "OFFLINE"
+	return msg
```

Comment:
Don't overwrite message without including its prior contents here, also call parent proc
I don't know what that will break but i dont think any other subsystem does that
```suggestion
/datum/controller/subsystem/bccm/stat_entry(msg)
	. = ..("[msg][is_active? "ACTIVE" : "OFFLINE]")
```

Reply:
you may also want to add something to stat for when the system is disabled